### ISSUE TITLE
onnx 推理最后一块不稳定的原因？

### ISSUE THREAD

**Issue by Wentao795** [FunAudioLLM/CosyVoice#422]
hi，感谢大佬的工作。请问，为什么onnx最后一个chunk会不稳定？大佬有相关的思路吗？或者说这个问题能被解决吗？期待给点思路，感谢~~

**Comment by aluminumbox** [comment]
last frame has dynamic time step, so the rtf is unstable, you can try pad it to same length with 0

**Comment by Wentao795** [comment]
抱歉，我之前的描述可能不太准确。
我的问题是是，这一步为什么会波动很大。output = self.estimator.run(None, ort_inputs)[0]
    def forward_estimator(self, x, mask, mu, t, spks, cond):
        if isinstance(self.estimator, torch.nn.Module):
            return self.estimator.forward(x, mask, mu, t, spks, cond)
        else:
            ort_inputs = {
                'x': x.cpu().numpy(),
                'mask': mask.cpu().numpy(),
                'mu': mu.cpu().numpy(),
                't': t.cpu().numpy(),
                'spks': spks.cpu().numpy(),
                'cond': cond.cpu().numpy()
            }
            start = time.time()
            output = self.estimator.run(None, ort_inputs)[0]
            print("one onnx inference need time:",time.time()-start,"input x  shape is:",x.size())
            return torch.tensor(output, dtype=x.dtype, device=x.device)

**Comment by Wentao795** [comment]
抱歉，我之前的描述可能不太准确。
我的问题是，这一步速度为什么会波动很大。
![image](https://github.com/user-attachments/assets/4107be66-8130-42d9-92c8-6015e82f176b)

**Comment by aluminumbox** [comment]
因为它最后一帧的时候时间步不确定

**Comment by Wentao795** [comment]
这是我跑的一句的测试log。
![image](https://github.com/user-attachments/assets/6c7e7443-08b1-4458-9806-ac5e23a4888c)
![image](https://github.com/user-attachments/assets/6126f7f9-d90b-4822-b432-4f138cf7b70b)

[cosyvoicelot.txt](https://github.com/user-attachments/files/17093346/cosyvoicelot.txt)

**Comment by Wentao795** [comment]
大佬，我还是不太明白，为什么是最后一帧。我的log显示并不是最后一次输出会耗时很长。~~
